src/twitch.ts

```diff
@@ -133,26 +128,40 @@ const twitch = {
   notLurkingAnymore(
     usernameOrSubmitter: string | Partial<QueueSubmitter>
   ): boolean {
-    let username: string;
+    lurkers.purgeStale(); // manually calling this because we are calling entries()
+    let username: string | undefined;
+    let displayName: string | undefined;
     if (typeof usernameOrSubmitter === "string") {
       username = usernameOrSubmitter;
+      displayName = usernameOrSubmitter;
     } else {
-      if (usernameOrSubmitter.login != null) {
-        username = usernameOrSubmitter.login;
+      if (usernameOrSubmitter.id != null) {
+        return lurkers.delete(usernameOrSubmitter.id);
+      }
+      if (usernameOrSubmitter.name != null) {
+        username = usernameOrSubmitter.name;
       } else if (usernameOrSubmitter.displayName != null) {
-        // best effort for now!
-        username = usernameOrSubmitter.displayName.toLowerCase();
+        displayName = usernameOrSubmitter.displayName;
       } else {
         // can not remove anyone with only `id` or no information
```

Comment:
Ah! Good catch, the comment is wrong.
Should be something like `Can not remove anyone with no id, name, and displayName`.
Since the function argument is a `Partial<QueueSubmitter>` one can pass in an empty object.